From 35a66f1b0e430d551853f4cb124c8d72900ed139 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Mon, 10 Feb 2025 14:44:45 +0100
Subject: [PATCH 792/841] refactor(st-ddr): remove DDR training area restore
 from driver

DDR training area restore should be called at platform level to
ensure having the same conditions as during the save operation.
This consistency is mandatory when encryption is activated on this
DDR training area.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 3328d6b13bcfd3455a9beb05d825172c5c2d43e4 refactor(st-ddr): remove DDR training area restore from driver]
Co-authored-by: Dirk Schoebel <dirk@liji.de>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: Ie7af3eb91ecb718875e74146801a1c40141d6db5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/435912
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/ddr/stm32mp1_ram.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/st/ddr/stm32mp1_ram.c b/drivers/st/ddr/stm32mp1_ram.c
index 3e9571217..9f4ec55e5 100644
--- a/drivers/st/ddr/stm32mp1_ram.c
+++ b/drivers/st/ddr/stm32mp1_ram.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2018-2023, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2018-2025, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
  */
@@ -121,9 +121,6 @@ static int stm32mp1_ddr_setup(void)
 			      uret);
 			panic();
 		}
-
-		/* Restore area overwritten by training */
-		stm32_restore_ddr_training_area();
 	} else {
 		size_t retsize;
 
-- 
2.47.2

